### ISSUE TITLE
Release Version 2.2.0

### ISSUE THREAD

**Issue by peterzhuamazon** [opensearch-project/OpenSearch#4017]
Release Version 2.2.0

This is a component issue for __2.2.0__.
Coming from [opensearch-build#2271](https://github.com/opensearch-project/opensearch-build/issues/2271). Please follow the following checklist.
__Please refer to the DATES in that post__.

<details><summary>How to use this issue</summary>
<p>

## This Component Release Issue

This issue captures the state of the OpenSearch release, on component/plugin level; its assignee is responsible for driving the release. Please contact them or @mention them on this issue for help.
Any release related work can be linked to this issue or added as comments to create visiblity into the release status.

## Release Steps

There are several steps to the release process; these steps are completed as the whole component release and components that are behind present risk to the release. The component owner resolves the tasks in this issue and communicate with the overall release owner to make sure each component are moving along as expected.

Steps have completion dates for coordinating efforts between the components of a release; components can start as soon as they are ready far in advance of a future release. The most current set of dates is on the overall release issue linked at the top of this issue.

## The Overall Release Issue

Linked at the top of this issue, the overall release issue captures the state of the entire OpenSearch release including references to this issue, the release owner which is the assignee is responsible for communicating the release status broadly.  Please contact them or @mention them on that issue for help.

## What should I do if my plugin isn't making any changes?

If including changes in this release, increment the version on `2.0` branch to `2.2.0` for Min/Core, and `2.2.0.0` for components. Otherwise, keep the version number unchanged for both.

</p>
</details>

### Preparation

- [x] Assign this issue to a release owner.
- [x] Update Apache Lucene to 9.3.0
- [x] Update OpenJDK (July 2022 Patch releases)
- [x] Finalize scope and feature set and update [the Public Roadmap](https://github.com/orgs/opensearch-project/projects/1).
- [x] All the tasks in this issue have been reviewed by the release owner.
- [x] Create, update, triage and label all features and issues targeted for this release with `v2.2.0`.
- [x] Cut 2.2 branch
- [x] Bump 2.x to version 2.3.0 and update BWC versions
- [x] Update BWC version in main 3.0.0

### CI/CD

- [x] All code changes for `2.2.0` are complete.
- [x] Ensure working and passing CI.
- [x] Check that this repo is included in the [distribution manifest](https://github.com/opensearch-project/opensearch-build/blob/main/manifests/2.2.0).

### Pre-Release

- [x] Update to the `2.2.0` release branch in the [distribution manifest](https://github.com/opensearch-project/opensearch-build/blob/main/manifests/2.2.0).
- [x] Increment the version on the parent branch to the next development iteration.
- [x] Gather, review and publish release notes following the [rules](https://github.com/opensearch-project/opensearch-plugins/blob/main/RELEASE_NOTES.md) and back port it to the release branch.[git-release-notes](https://github.com/ariatemplates/git-release-notes) may be used to generate release notes from your commit history.
- [x] Confirm that all changes for `2.2.0` have been merged.
- [x] Add this repo to the [manifest](https://github.com/opensearch-project/opensearch-build/blob/main/manifests/) for the next developer iteration.

### Release Testing

- [x] Find/fix bugs using latest tarball and docker image provided in parent release issue and update the release notes if necessary.
- [x] Code Complete: Test within the distribution, ensuring integration, backwards compatibility, and performance tests pass.
- [x] Sanity Testing: Sanity testing *and* fixing of critical issues found.
- [x] File issues for all intermittent test failures.

### Release

- [x] Complete [documentation](https://github.com/opensearch-project/documentation-website).
- [x] Verify all issued labeled for this release are closed or labelled for the next release.

### Post Release

- [ ] Prepare for an eventual security fix development iteration by incrementing the version on the release branch to the next eventual patch version.
- [ ] Add this repo to the [manifest](https://github.com/opensearch-project/opensearch-build/blob/main/manifests/) of the next eventual security patch version.
- [ ] [Suggest improvements](https://github.com/opensearch-project/opensearch-build/issues/new) to [this template](https://github.com/opensearch-project/opensearch-build/blob/main/meta/templates/releases/release_template.md).
- [ ] Conduct a retrospective, and publish its results.

**Comment by kartg** [comment]
* Apache Lucene has been updated to 9.3.0 - https://github.com/opensearch-project/OpenSearch/pull/4088
* OpenJDK has been updated with backports to `2.x`, `2.0` and `2.1` - https://github.com/opensearch-project/OpenSearch/pull/4023

All thanks to @reta 🏅

**Comment by kartg** [comment]
`2.2` branch created - https://github.com/opensearch-project/OpenSearch/tree/2.2

I'll work on bumping `2.x` to `2.3.0` next

**Comment by kartg** [comment]
* `2.x` has been bumped to `2.3.0` and BWC versions have been updated in `main`. 
* The manifest file in `opensearch-build` has been updated to point to the `2.2` branch rather than `2.x`
* I'm currently triaging [all open issues tagged with v2.2.0](https://github.com/opensearch-project/OpenSearch/labels/v2.2.0)
  * Aside from the issues above, https://github.com/opensearch-project/OpenSearch/pull/2293 is also aiming to make it to `2.2.0`. This was identified via [the project roadmap](https://github.com/orgs/opensearch-project/projects/1)

**Comment by kartg** [comment]
There are [no more open issues tagged with v2.2.0](https://github.com/opensearch-project/OpenSearch/labels/v2.2.0). #2293 has also been backported to 2.2 - https://github.com/opensearch-project/OpenSearch/pull/4141

**Comment by kartg** [comment]
Generating release notes as outlined here - https://github.com/opensearch-project/OpenSearch/issues/789#issuecomment-1179436917

* `2.x` was bumped to `2.2.0` on Jun 24 - https://github.com/opensearch-project/OpenSearch/commit/718ea66c2c3f28ad1d432ec4943dea3ae651efec
* Generated rough draft of release notes, with an in-line `sed` command to replace PR ids with the full Github URL:
```
git log --pretty=format:"* %s" --since=6-24-2022 | sed -E "s|(\(#([0-9]*)\))|([#\2](https://github.com/opensearch-project/opensearch/pull/\2))|g"
```
* Note that I'm generating these notes off the `2.2` branch and not `main`. I'll eventually cherry-pick this change to `main`
* Manual organize the release notes into [the categories defined here](https://github.com/opensearch-project/opensearch-plugins/blob/main/RELEASE_NOTES.md#categories).

**Comment by reta** [comment]
@kartg any chance to get https://github.com/opensearch-project/OpenSearch/pull/4145 in?

**Comment by peterzhuamazon** [comment]
> @kartg any chance to get #4145 in? sorry for dragging the release ...

Hi @reta as shown in the schedule of this post, it is code freeze today so normally we would against merging any code to 2.2 anymore unless it is critical function breaking or security related.
However I was resolving a bug earlier as the build pipeline is not pass yet, so I will accept this change this time. 😄 

Thanks.

**Comment by bbarani** [comment]
@kartg @anasalkouz  I am closing this issue since we released 2.2.0 already but I would request you to keep this issue up to date during release and close it on time going forward. Make sure to add any retrospective learning on this issue as well.
